src/base/emu-i386/simx86/codebuf.c

```diff
@@ -93,14 +94,15 @@ void FreeGenCodeBuf(void *ptr)
     mspace_free(cspace, ptr);
 }
 
-unsigned char *GetGenCodeBuf(const unsigned char *eip)
+unsigned char *GetGenCodeBuf(uintptr_t eip)
 {
-    assert(eip >= xbase && eip < xbase + CODEBUF_SZ);
-    return wbase + (eip - xbase);
+    unsigned char *p = (unsigned char *)eip - xoffset;
```

Comment:
I think its UB though.
You are not allowed to do
pointer arithmetic across
the unrelated objects in
memory.